evalml/pipelines/components/transformers/preprocessing/polynomial_detrender.py

```diff
@@ -112,7 +112,7 @@ def inverse_transform(self, y):
         """
         if y is None:
             raise ValueError("y cannot be None for PolynomialDetrender!")
-        y_dt = infer_feature_types(y)
-        y_t = self._component_obj.inverse_transform(y_dt)
-        y_t = infer_feature_types(pd.Series(y_t, index=y_dt.index))
+        y_ww = infer_feature_types(y)
```

Comment:
Is part of the underlying problem here that `infer_feature_types()` should be preserving the index?  Should we do it in there?

Reply:
@chukarsten Unfortunately, I think regardless of our `infer_feature_types` behavior (which I think should preserve indices but I'm not positive), the issue is that a lot of third party libraries might return data such as np.arrays or new pandas objects that don't have the indices attached to them. 🥲